packages/venia-concept/src/actions/checkout/asyncActions.js

```diff
@@ -215,16 +273,35 @@ export function formatAddress(address = {}, countries = []) {
     };
 }
 
-async function clearAddress() {
-    return storage.removeItem('address');
+/**
+ * When submitting addresses to the Magento backend, we must remove all
+ * properties it doesn't recognize because it will reject the submission.
+ *
+ * @param {Object} address - an address object that may have extra properties.
+ */
+export function removeInvalidKeysFromAddress(
+    address,
+    invalidKeys = INVALID_ADDRESS_KEYS
+) {
+    const validAddress = {};
+    const keysToKeep = Object.keys(address).filter(
+        key => !invalidKeys.includes(key)
+    );
+    keysToKeep.forEach(key => (validAddress[key] = address[key]));
+
+    return validAddress;
 }
 
-async function retrieveAddress() {
-    return storage.getItem('address');
+async function clearBillingAddress() {
+    return storage.removeItem('billing_address');
 }
 
-async function saveAddress(address) {
-    return storage.setItem('address', address);
+async function retrieveBillingAddress() {
+    return storage.getItem('billing_address');
```

Comment:
If you keep the `sameAsShippingAddress` stuff around you could move the `removeInvalidKeysFromAddress` step here. Or you could scrub the boolean from the initial setting below:

```js
if (address.isSameAsShippingAddress) {
  return storage.setItem(storage.getItem('shipping_address')
}
```

etc.